Sources/CoffeeAddictsLibrary/Helpers/CoffeeShopsCSVDownloader.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  File.swift
+//  
+//
+//  Created by Mihai Garda Popescu on 23.07.2021.
+//
+
+import Foundation
+
+typealias CSVResult = (String?, ParsingStatus) -> Void
+
+class CoffeeShopsCSVDownloader {
+    static func downloadCSVFileFromURLString(_ urlString: String, completion: @escaping CSVResult) {
+        let sema = DispatchSemaphore(value: 0)
+
+        let documentsUrl:URL =  (FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first as URL?)!
+        let destinationFileUrl = documentsUrl.appendingPathComponent(Constants.localCSVFileName)
+        
+        do {
+            try FileManager.default.removeItem(at: destinationFileUrl)
+        } catch {
+            completion(nil, .CSVFileDownloadError)
+        }
+
+        let fileURL = URL(string: urlString)
+        
+        let sessionConfig = URLSessionConfiguration.default
+        let session = URLSession(configuration: sessionConfig)
+     
+        let request = URLRequest(url:fileURL!)
+        
+        let task = session.downloadTask(with: request) { (tempLocalUrl, response, error) in
+            if let tempLocalUrl = tempLocalUrl, error == nil {
+                do {
+                    try FileManager.default.copyItem(at: tempLocalUrl, to: destinationFileUrl)
+                } catch (let writeError) {
+                    print("Error creating a file \(destinationFileUrl) : \(writeError)")
+                    completion(nil, .CSVFileDownloadError)
+                }
+                
+                if let dir = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first {
```

Comment:
If I understand correctly, here you have the same url as you get on `documentsUrl` ? Can't you use that one?

Reply:
Done